### PR TITLE
[risk=no][RW-1459]Must be project owner to edit workspace or research purpose

### PR DESCRIPTION
**Pull request by jacmrob**

### CODE REVIEW

File: ui/src/app/views/workspace-nav-bar/component.html [hunk 1]
```diff
@@ -35,12 +35,12 @@
...
         </button>
         <span role="tooltip" aria-haspopup="true" class="tooltip tooltip-sm tooltip-top-left">
-          <button clrDropdownItem class="btn" [disabled]="!writePermission" routerLink="/workspaces/{{wsNamespace}}/{{wsId}}/edit"
+          <button clrDropdownItem class="btn" [disabled]="!ownerPermission" routerLink="/workspaces/{{wsNamespace}}/{{wsId}}/edit"
                       routerLinkActive="active">
             <clr-icon shape="pencil" class="icons"></clr-icon>
             Edit
           </button>
-          <span *ngIf="!writePermission" class="tooltip-content">Requires Write Permission</span>
+          <span *ngIf="!ownerPermission" class="tooltip-content">Requires Owner Permission</span>
```
Comment (jacmrob) [inline ui/src/app/views/workspace-nav-bar/component.html]: question, maybe for @lstrano-broad  - do we want the tooltip language to be consistent across this dropdown?  for clone and edit we say "Requires Owner Permission", but for share we say "Owner access is required to share the workspace"

**Comment by calbach** [comment]
I just looked at the code and got equally confused. "Research Purpose" in the UI is actually a header for the workspace description + the actual researchPurpose from the workspace model. The latter cannot be updated, but description can be. The API changes you made should cover this.

Should the workspace edit page also be updated in this PR? (theoretically there should be no more links to it anyways, but..)

**Comment by jacmrob** [comment]
i'm not sure.  anyone have thoughts?  @blrubenstein @UrsaStutsman @rushtong @NehaBroad ?

File: ui/src/app/views/workspace-nav-bar/component.html (hunk 1 above)
Comment (lstrano-broad) [reply ui/src/app/views/workspace-nav-bar/component.html]: ok to be consistent

**Comment by jacmrob** [comment]
update: we discussed, this is intended behavior for now, and the edit research purpose button is probably going to be modified in the future.  for now API changes are covered.

**Pull request merged**